File: app/src/main/java/me/eagzzycsl/darkroom/db/MySQLiteHelper.java
package me.eagzzycsl.darkroom.db;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;


class MySQLiteHelper extends SQLiteOpenHelper implements AboutDb {

    MySQLiteHelper(Context context, String name, SQLiteDatabase.CursorFactory factory, int version) {
        super(context, name, factory, version);

    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        final String sql_create =
                "CREATE TABLE IF NOT EXISTS " + TABLE_Naughty.TABLE_NAME + "("
                        + TABLE_Naughty.FILED.ID + " integer primary key autoincrement, "
                        + TABLE_Naughty.FILED.pkgName + " text);";
        db.execSQL(sql_create);
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {

    }

}

File: app/src/main/java/me/eagzzycsl/darkroom/db/MySQLiteHelper.kt
package me.eagzzycsl.darkroom.db

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper


class MySQLiteHelper(context: Context, name: String, factory: SQLiteDatabase.CursorFactory?, version: Int) : SQLiteOpenHelper(context, name, factory, version) {

    override fun onCreate(db: SQLiteDatabase) {
        val sqlCreate = (
                "CREATE TABLE IF NOT EXISTS ${AboutDb.NaughtyTable.TABLE_NAME}("
                        + "${AboutDb.NaughtyTable.FILED.ID} integer primary key autoincrement, "
                        + "${AboutDb.NaughtyTable.FILED.isNaughty} integer DEFAULT 1, "
                        + "${AboutDb.NaughtyTable.FILED.freezeNotifyTimes} integer DEFAULT 0, "
                        + "${AboutDb.NaughtyTable.FILED.pkgName}  text);")
        db.execSQL(sqlCreate)
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        if (oldVersion == 1) {
            db.execSQL(
                    "alter table ${AboutDb.NaughtyTable.TABLE_NAME} add column "
                            + "${AboutDb.NaughtyTable.FILED.isNaughty} integer DEFAULT 1"
            )
            db.execSQL(
                    "alter table ${AboutDb.NaughtyTable.TABLE_NAME} add column "
                            + "${AboutDb.NaughtyTable.FILED.freezeNotifyTimes} integer DEFAULT 0"
            )

        }
    }

}